In [10]:
from sklearn.metrics import precision_score,recall_score,f1_score,confusion_matrix,classification_report,accuracy_score
from sklearn.datasets import make_classification
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
import pandas as pd
import numpy as np

In [22]:
X,y=make_classification(n_samples=10000,n_features=10,n_classes=2,weights=[0.99,0.01],random_state=1)

In [23]:
type(X)

numpy.ndarray

In [24]:
X.shape

(10000, 10)

In [25]:
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.3,random_state=10)

In [26]:
X_train.shape,y_test.shape

((7000, 10), (3000,))

In [27]:
model=LogisticRegression()

In [28]:
model.fit(X_train,y_train)

c:\users\sudhe\appdata\local\programs\python\python37\lib\site-packages\sklearn\linear_model\logistic.py:432: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)


LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=100,
                   multi_class='warn', n_jobs=None, penalty='l2',
                   random_state=None, solver='warn', tol=0.0001, verbose=0,
                   warm_start=False)

In [41]:
y_predict=model.predict(X_test)

In [30]:
precision_score(y_test,y_predict)

0.9166666666666666

In [31]:
accuracy_score(y_test,y_predict)

0.9876666666666667

In [32]:
recall_score(y_test,y_predict)

0.23404255319148937

In [33]:
f1_score(y_test,y_predict)

0.3728813559322034

In [34]:
confusion_matrix(y_test,y_predict)

array([[2952,    1],
       [  36,   11]], dtype=int64)

In [35]:
print(classification_report(y_test,y_predict))

              precision    recall  f1-score   support

           0       0.99      1.00      0.99      2953
           1       0.92      0.23      0.37        47

    accuracy                           0.99      3000
   macro avg       0.95      0.62      0.68      3000
weighted avg       0.99      0.99      0.98      3000



In [36]:
y.sum()

151

# only 151 1's of 10k points
# so 1 is minority and we got scores for the same when we used precision_score,recall_score,f1_score
# using classification report we got for both
# from these we can say it is not a good model to classify

In [37]:
from sklearn.ensemble import RandomForestClassifier

In [136]:
rfc=RandomForestClassifier(n_estimators=30,random_state=10,min_samples_split=5)

In [137]:
rfc.fit(X_train,y_train)

RandomForestClassifier(bootstrap=True, class_weight=None, criterion='gini',
                       max_depth=None, max_features='auto', max_leaf_nodes=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=1, min_samples_split=5,
                       min_weight_fraction_leaf=0.0, n_estimators=30,
                       n_jobs=None, oob_score=False, random_state=10, verbose=0,
                       warm_start=False)

In [138]:
y_predict2=rfc.predict(X_test)

In [139]:
confusion_matrix(y_test,y_predict2)

array([[2946,    7],
       [  26,   21]], dtype=int64)

In [140]:
print(classification_report(y_test,y_predict2))

              precision    recall  f1-score   support

           0       0.99      1.00      0.99      2953
           1       0.75      0.45      0.56        47

    accuracy                           0.99      3000
   macro avg       0.87      0.72      0.78      3000
weighted avg       0.99      0.99      0.99      3000



# this model is better than previous

In [ ]:
from sklearn.ensemble 